### ISSUE TITLE
Upgrade to .NET Core 3

### ISSUE THREAD

**Issue by JasonGT** [JasonGT/NorthwindTraders#91]

**Comment by JasonGT** [comment]
This is curently blocked, pending .NET Core 3 upgrade of:
* NSwag
* FluentValidation

**Comment by JosKrause** [comment]
@JasonGT 

If and when this upgrade happens would you consider creating dedicated branches for 2.x vs 3.x?

**Comment by rafalschmidt97** [comment]
FluentValidation news:

> I've pushed [FluentValidation.AspNetCore 8.5.0-preview1](https://www.nuget.org/packages/FluentValidation.AspNetCore/8.5.0-preview1) to nuget. This supports ASP.NET Core 3 preview 5.

But Nswag is still not ready yet ([issue](https://github.com/RicoSuter/NSwag/issues/1961)) 🚀

**Comment by JasonGT** [comment]
> FluentValidation [news](https://github.com/JeremySkinner/FluentValidation/issues/976):
> 
> > I've pushed [FluentValidation.AspNetCore 8.5.0-preview1](https://www.nuget.org/packages/FluentValidation.AspNetCore/8.5.0-preview1) to nuget. This supports ASP.NET Core 3 preview 5.
> 
> But Nswag is still not ready yet ([issue](https://github.com/RicoSuter/NSwag/issues/1961)) 🚀

Awesome, thanks for the heads up. :)

**Comment by JasonGT** [comment]
> @JasonGT
> 
> If and when this upgrade happens would you consider creating dedicated branches for 2.x vs 3.x?

Will do

**Comment by JasonGT** [comment]
I've created new branches [dotnetcore22](https://github.com/JasonGT/NorthwindTraders/tree/dotnetcore22) and [dotnetcore3](https://github.com/JasonGT/NorthwindTraders/tree/dotnetcore3).

The upgrade has been completed on the dotnetcore3 branch, using development builds of .NET Core 3 (currently 3.0.0-rc1-19425-03). I choose to use the development builds since EF Core issues were encountered in .NET Core 3 Preview 8.

I didn't encounter any issues with FluentValidation or MediatR.

For NSwag, I had to use a different approach since NSwag.MSBuild doesn't yet support .NET Core 3 (see ReadMe for workaround).

I'm planning to keep this branch until we get the first release candidate, at which point if all is good I'll merge into master. The dotnetcore22 branch will remain, but not likely to see any updates going forward.

Let me know if you have any feedback or suggestions.
